### PR TITLE
ISPN-6002 ISPN-6003 Minor performance-related changes

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-6002
https://issues.jboss.org/browse/ISPN-6003

Pretty minor changes, easy to make but hard to measure a difference in performance after them...

### CODE REVIEW

File: core/src/main/java/org/infinispan/interceptors/CacheWriterInterceptor.java [hunk 1]
```diff
@@ -236,11 +237,12 @@ public Object visitWriteOnlyKeyCommand(InvocationContext ctx, WriteOnlyKeyComman
             if (entry != null) {
                if (entry.isRemoved()) {
                   boolean resp = persistenceManager.deleteFromAllStores(key, BOTH);
-                  if (getLog().isTraceEnabled()) getLog().tracef("Removed entry under key %s and got response %s from CacheStore", key, resp);
+                  if (trace) getLog().tracef("Removed entry under key %s and got response %s from CacheStore", key, resp);
                } else if (entry.isChanged()) {
                   storeEntry(ctx, key, command);
                }
             }
+            break;
```
Comment (wburns) [inline core/src/main/java/org/infinispan/interceptors/CacheWriterInterceptor.java]: heh, good catch

File: core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java [hunk 2]
```diff
@@ -92,20 +94,22 @@ private DecoratedCache(CacheImpl<K, V> delegate, ClassLoader classLoader, EnumSe
       if (flags == null || flags.length == 0)
          return this;
       else {
-         final List<Flag> flagsToAdd = Arrays.asList(flags);
-         if (this.flags != null && this.flags.containsAll(flagsToAdd)) {
+         boolean containsAll = false;
```
Comment (wburns) [inline core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java]: You could do this assignment in an else block after line 105 as well.

**Comment by wburns** [comment]
Seems fine to me, just very minor comment.  However it does appear jcache failed to compile

http://ci.infinispan.org/viewLog.html?buildId=32853&buildTypeId=bt9&tab=buildLog&filter=err

File: jcache/embedded/src/main/java/org/infinispan/jcache/embedded/JCache.java [hunk 3]
```diff
@@ -146,7 +147,7 @@ public void clear() {
    public boolean containsKey(final K key) {
       checkNotClosed();
 
-      if (log.isTraceEnabled())
+      if (trace) {
```
Comment (wburns) [inline jcache/embedded/src/main/java/org/infinispan/jcache/embedded/JCache.java]: Extra right brace

File: jcache/embedded/src/main/java/org/infinispan/jcache/embedded/JCache.java [hunk 4]
```diff
@@ -271,7 +272,7 @@ public String getName() {
       // verify that under contended access, one of the threads should "wait"
       // for the other, hence the use locks.
 
-      if (log.isTraceEnabled())
+      if (trace) {
```
Comment (wburns) [inline jcache/embedded/src/main/java/org/infinispan/jcache/embedded/JCache.java]: Extra right brace

**Comment by danberindei** [comment]
Oops, sorry about the JCache errors... way too much copy-paste for one day :)

**Comment by wburns** [comment]
Pulling...

**Comment by wburns** [comment]
Integrated into master, thanks @danberindei !

**Pull request closed without merging**